contracts/strategies/liquity/LiquityLQTYStakingStrategy.sol

```diff
@@ -0,0 +1,311 @@
+     */
+    function setSell(bool _sell) public onlyGovernance {
+        _setSell(_sell);
+    }
+
+    /**
+     * @dev Set the minimum amount of DODO needed to trigger a sale (for a liquidation).
+     */
+    function setSellFloor(uint256 _sellFloor) public onlyGovernance {
+        setSellFloor(_sellFloor);
+    }
+
+    function finalizeUpgrade() external onlyGovernance {
+        _finalizeUpgrade();
+    }
+
+    // ---------------- Internal methods ----------------
+
+    function rewardPoolBalance() internal view returns (uint256 balance) {
+        balance = ILQTYStaking(rewardPool()).stakes(address(this));
+    }
+
+    function claimRewards() internal {
+        if (rewardPoolBalance() > 0) {
+            ILQTYStaking(rewardPool()).unstake(0);
+        }
+    }
+
+    function enterRewardPool() internal {
+        uint256 currentBalance = IERC20(underlying()).balanceOf(address(this));
+        ILQTYStaking(rewardPool()).stake(currentBalance);
+    }
+
+    function exitRewardPool() internal {
+        uint256 staked = rewardPoolBalance();
+        if (staked > 0) {
+            ILQTYStaking(rewardPool()).unstake(staked);
+        }
+    }
+
+    /**
+     * @dev Stakes everything the strategy holds into the reward pool.
+     */
+    function investAllUnderlying() internal onlyNotPausedInvesting {
+        // This check is needed for avoiding reverts in case we stake 0
+        if (IERC20(underlying()).balanceOf(address(this)) > 0) {
+            enterRewardPool();
+        }
+    }
+
+    function _liquidateReward() internal {
+        // Profits can be disabled for possible simplified and rapid exit
+        if (!sell()) {
+            emit ProfitsNotCollected(sell(), true);
+            return;
+        }
+
+        uint256 lusdBalance = IERC20(lusd).balanceOf(address(this));
+        uint256 ethBalance = address(this).balance;
+
+        // Liquidate the LUSD rewards to USDC
+        if (lusdBalance > 0) {
+            IERC20(lusd).safeApprove(lusd3CrvPair, 0);
+            IERC20(lusd).safeApprove(lusd3CrvPair, lusdBalance);
+            ICurveMetapool(lusd3CrvPair).exchange_underlying(
+                0,
+                2,
+                lusdBalance,
+                1
+            );
+        }
+
+        // Liquidate the ETH rewards to USDC
+        if (ethBalance > 0) {
+            ISwapRouter(uniswapV3Router).exactInputSingle.value(ethBalance)(
+                ISwapRouter.ExactInputSingleParams(
+                    weth,
+                    usdc,
+                    500,
+                    address(this),
+                    block.timestamp,
+                    ethBalance,
+                    1,
+                    0
+                )
+            );
+        }
+
+        // At this point, both LUSD and ETH rewards were swapped for USDC,
+        // which we now handle as the reward token
+        uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
```

Comment:
Could we use WETH as a rewardToken() instead? This would reduce the number of hops, and it's what we commonly did for cases when there are multiple liquidations